src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -51,6 +51,11 @@ internal abstract class CSharpSemanticModel : SemanticModel
         /// </summary>
         internal abstract CSharpSyntaxNode Root { get; }
 
+
+        public override SemanticModel WithSuppressAccessChecks()
+        {
+            return new SyntaxTreeSemanticModel(Compilation, this.SyntaxTree, suppressAccessChecks: true);
```

Comment:
If the current model is already suppressing the checks, we should just return it.